examples/docker/gpt_neo_docker.py

```diff
@@ -0,0 +1,76 @@
+"""
+Pipline has support for creating FastAPI endpoints for your Pipelines out of the box.
+This file demonstrates this by creating a basic GPT-Neo Pipeline with a custom
+environment and deploying the docker image.
+
+Run this code in this directory:
+
+python gpt_neo_docker.py
+
+and then start the docker containers created:
+
+sudo docker compose up -d
+
+you can test the running images:
+
+
+curl --request POST \
+  --url http://localhost:5010/v2/run \
+  --header 'Content-Type: application/json' \
+  --data '{"pipeline_id":"gptneo", "data":"Hey there my name is"}'
+
+"""
+
+
+from pipeline import Pipeline, Variable, docker, pipeline_function, pipeline_model
+from pipeline.objects.environment import Environment
+
+
+@pipeline_model
+class PipelineGPTNeo:
+    def __init__(self):
+        self.model = None
+        self.tokenizer = None
+
+    @pipeline_function
+    def predict(self, input_data: str) -> str:
+        if self.model is None:
+            from transformers import GPT2Tokenizer, GPTNeoForCausalLM
+
+            self.model = GPTNeoForCausalLM.from_pretrained("EleutherAI/gpt-neo-125M")
+            self.tokenizer = GPT2Tokenizer.from_pretrained("EleutherAI/gpt-neo-125M")
```

Comment:
No preference from me, but why this style rather than the usual `load` function?